clients.rst

```diff
@@ -1,11 +1,17 @@
 Clients & Adapters
 ==================
 
-There are clients implementing one of the HTTPlug interfaces directly,
-and adapter packages that implement the interface and forward the calls to HTTP clients not implementing the interface.
+There are two types of libraries you can use to send HTTP messages; clients and adapters. A client implements the
+``HttpClient`` and/or the ``HttpAsyncClient`` interfaces directly. A client adapter is a class implementing the
+interface and forward the calls to a HTTP client not implementing the interface. (See `Adapter pattern`_ on Wikipedia).
 
-.. _`php-http/client-implementation`: https://packagist.org/providers/php-http/client-implementation
-.. _`php-http/async-client-implementation`: https://packagist.org/providers/php-http/async-client-implementation:
+.. note::
+
+    All clients and adapters comply with `Liskov substitution principle`_ which means that you can easily change one
+    for another without any side effects.
+
+.. _`Adapter pattern`: https://en.wikipedia.org/wiki/Adapter_pattern
+.. _`Liskov substitution principle`: https://en.wikipedia.org/wiki/Liskov_substitution_principle
```

Comment:
I usually have likes that I use more than once at the bottom. But I'll change and make sure I put all links in the bottom in the future.